### PR TITLE
More Python 3 fixes.

### PR DESCRIPTION
**Pull request by hannosch**
Special call out to `bytes(3)`, which produces `'3'` in Python 2 and `b'\x00\x00\x00'` in Python 3. I also like Python 3.5 and bringing back % formatting for bytes :(

This fixes another round of test failures. Logging in via basic auth is still broken, and request/response issues around headers and the body being str/bytes.

Python 3.4: 1118 tests, 45 failures, 36 errors and 1 skipped in 8.470 seconds.

### CODE REVIEW

File: src/OFS/tests/testRanges.py [hunk 1]
```diff
@@ -186,15 +186,20 @@ def expectMultipleRanges(self, range, sets, draft=0):
...
+            self.assertEqual(rsp.getHeader('content-length'), str(len(body)))
 
-        # Decode the multipart message
-        bodyfile = io.BytesIO('Content-Type: %s\n\n%s' % (
-            rsp.getHeader('content-type'), body))
-        partmessages = [part
-                        for part in email.message_from_file(bodyfile).walk()]
+        # Decode the multipart message and force a latin-1 encoding,
+        # revert that later after the email was parsed
+        bodyfile = io.StringIO(
```
Comment (tseaver) [inline src/OFS/tests/testRanges.py]: Should this be `BytesIO`?

File: src/Testing/ZopeTestCase/ZopeLite.py [hunk 2]
```diff
@@ -28,14 +28,16 @@
...
 
 from Testing.ZopeTestCase import layer
 
 # Allow code to tell it is run by the test framework
 os.environ['ZOPETESTCASE'] = '1'
 
 # Increase performance on MP hardware
-sys.setcheckinterval(2500)
+if PY2:
+    sys.setcheckinterval(2500)
```
Comment (tseaver) [inline src/Testing/ZopeTestCase/ZopeLite.py]: Ugh, this is a nasty import time side effect.

File: src/OFS/tests/testRanges.py (hunk 1 above)
Comment (hannosch) [inline src/OFS/tests/testRanges.py]: I thought so too. But when I tried the ` email.message_from_file` call just below with binary it blew up. Apparently that wants text. That's why I ended up with the forced latin-1 decode, get_payload encode trick.

File: src/Testing/ZopeTestCase/ZopeLite.py (hunk 2 above)
Comment (hannosch) [inline src/Testing/ZopeTestCase/ZopeLite.py]: Yeah, to be fair, this is in a test-only module and has been there forever. I've just disabled it for PY3, as that only has the switchinterval and the default value for it is generally good.

**Pull request merged**